### PR TITLE
Fix "failed to create head node" issue

### PR DESCRIPTION
**Pull request by edoakes**
## Why are these changes needed?

`ray up` was sometimes failing with "Failed to create head node" when seemingly nothing was wrong. This was caused by creating the node, then immediately checking if it existed or not and failing if not. However, the node might have been created properly but just not yet show up in `boto3` API calls (or similar for other providers).

## Checks

- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [x] I've included any doc changes needed for https://ray.readthedocs.io/en/latest/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failure rates at https://ray-travis-tracker.herokuapp.com/.

### CODE REVIEW

**Review by pcmoritz — APPROVED** [review]
LGTM! Consider raising an exception that is more specific, e.g. https://docs.python.org/3/library/exceptions.html#RuntimeError or maybe an autoscaler specific one?

**Comment by pcmoritz** [comment]
Ah, actually, do you want to do the time.sleep after the break? Seems a bit wasteful to always sleep, even if the call already succeeded.

File: python/ray/autoscaler/commands.py [hunk 1]
```diff
@@ -199,9 +199,17 @@ def get_or_create_head_node(config, config_file, no_restart, restart_only, yes,
                 config["cluster_name"])
             provider.create_node(config["head_node"], head_node_tags, 1)
 
+        start = time.time()
```
Comment (mehrdadn) [inline python/ray/autoscaler/commands.py]: You want to use a monotonic clock to measure the passage of time, not a wall clock.
You can do this via `timeit.default_timer` in Python.

File: python/ray/autoscaler/commands.py [hunk 2]
```diff
@@ -199,9 +199,16 @@ def get_or_create_head_node(config, config_file, no_restart, restart_only, yes,
                 config["cluster_name"])
             provider.create_node(config["head_node"], head_node_tags, 1)
 
-        nodes = provider.non_terminated_nodes(head_node_tags)
-        assert len(nodes) == 1, "Failed to create head node."
-        head_node = nodes[0]
+        start = time.time()
+        head_node = None
+        while True:
+            if time.time() - start > 5:
```
Comment (ericl) [inline python/ray/autoscaler/commands.py:205]: cc @AmeerHajAli we should probably change this to 30 seconds or something.

**Pull request merged**